### ISSUE TITLE
Pio home crash at initial install 

### ISSUE THREAD

**Issue by xsbear** [pioarduino/platform-espressif32#80]
Description of problem
Leave a comment...

BEFORE SUBMITTING, PLEASE SEARCH FOR DUPLICATES IN

https://github.com/platformio/platformio-vscode-ide/issues%3Fq=is%3Aissue
%23 Configuration

VSCode: 1.96.0
PIO IDE: v1.0.4
System: Darwin, 24.2.0, arm64

%23 Exception

Error: TypeError: Traceback (most recent call last):
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/platformio/__main__.py", line 103, in main
  cli()  %23 pylint: disable=no-value-for-parameter
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
  return self.main(*args, **kwargs)
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/click/core.py", line 1078, in main
  rv = self.invoke(ctx)
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/platformio/cli.py", line 85, in invoke
  return super().invoke(ctx)
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
  return _process_result(sub_ctx.command.invoke(sub_ctx))
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
  return ctx.invoke(self.callback, **ctx.params)
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/click/core.py", line 783, in invoke
  return __callback(*args, **kwargs)
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/platformio/home/cli.py", line 92, in cli
  run_server(
File "/Users/didi/.platformio/penv/lib/python3.9/site-packages/platformio/home/run.py", line 66, in run_server
  if not os.path.isdir(contrib_dir):
File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/genericpath.py", line 42, in isdir
  st = os.stat(s)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType

============================================================

An unexpected error occurred. Further steps:

* Verify that you have the latest version of PlatformIO using
`python -m pip install -U platformio` command

* Try to find answer in FAQ Troubleshooting section
https://docs.platformio.org/page/faq/index.html

* Report this problem to the developers
https://github.com/platformio/platformio-core/issues

============================================================

  at /Users/didi/.vscode/extensions/pioarduino.pioarduino-ide-1.0.4/node_modules/pioarduino-node-helpers/dist/index.js:1:10938
  at ChildProcess.d (/Users/didi/.vscode/extensions/pioarduino.pioarduino-ide-1.0.4/node_modules/pioarduino-node-helpers/dist/index.js:1:4560)
  at ChildProcess.emit (node:events:518:28)
  at ChildProcess.emit (node:domain:489:12)
  at maybeClose (node:internal/child_process:1104:16)
  at Socket.<anonymous> (node:internal/child_process:456:11)
  at Socket.emit (node:events:518:28)
  at Socket.emit (node:domain:489:12)
  at Pipe.<anonymous> (node:net:343:12)

**Comment by Jason2866** [comment]
Follow up of https://github.com/pioarduino/platform-espressif32/issues/70#issuecomment-2548848364

**Comment by probonopd** [comment]
Also running into this. Is there any workaround?

**Comment by Jason2866** [comment]
Fixed nearly. A full fix is quiet complicated. PIO home is integrated in Platformio registry and very early loaded. Without the registry a full fix needs a lot of changes in pio core.
After initial install it is needed to load an existing pioarduino espressif project. After this is loaded and compiled one time PIOHome will work to open new / empty projects.

To test:
- Uninstall pioarduino plugin
-  Close VSC
- Delete hidden folder `.platformio`
- Start VSC
- Install pioarduino, follow instructions
- Now open an existing project and hit compile
- Close Project
- Now open "PIO Home"

**Comment by xsbear** [comment]
> Fixed nearly. A full fix is quiet complicated. PIO home is integrated in Platformio registry and very early loaded. Without the registry a full fix needs a lot of changes in pio core. After initial install it is needed to load an existing pioarduino espressif project. After this is loaded and compiled one time PIOHome will work to open new / empty projects.
> 
> To test:
> 
> * Uninstall pioarduino plugin
> * Close VSC
> * Delete hidden folder `.platformio`
> * Start VSC
> * Install pioarduino, follow instructions
> * Now open an existing project and hit compile
> * Close Project
> * Now open "PIO Home"

Follow these steps，It works！

**Comment by Jason2866** [comment]
It is now completely fixed. After a fresh install Pio Home is working without any extra steps

**Comment by tschissler** [comment]
I also had issues to uninstall PlatformIO 
Each time I restarted VSCode, the PlatformIo extension got reinstalled.
I was able to fix that by deleting the platformio folder from C:\Users\<User>\.vscode\extensions and removing the entry from the extensions.json in the same folder.

**Comment by mhaberler** [comment]
I'm sharing an observation in the hope it might help someone else.

original situation: 
- MacOS Arm64, Framework Python 3.11 from https://www.python.org/downloads/macos/
- pioarduino working on a project with Arduino 3.2

Since 3.11 is outdated I decided to upgrade, and reinstall everything from scratch:
- remove `/Applications/Python 3.11` completely
- reinstall Python 3.13 from https://www.python.org/downloads/macos/
- NB this will change $HOME/.zprofile to read:
`````
# Setting PATH for Python 3.13
# The original version is saved in .zprofile.pysave
PATH="/Library/Frameworks/Python.framework/Versions/3.13/bin:${PATH}"
export PATH
`````
- make sure the right Python version comes up at the command line:

`````
MacBookPro ~ $ which python3
/Library/Frameworks/Python.framework/Versions/3.13/bin/python3 # looks good
`````
- next: `rm -rf .platformio`
- run VSCode from command line: `code`
- uninstall `pioarduino` extension
- restart VSCode
- install `pioarduino` extension  - all fine so far
- now open an existing project which used to build - here's where the fun started: during package download, I got a Python splat mentioning `ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1028)`, eventually followed by:

`````
  File "<string>", line 6, in <module>
    print(json.dumps(load_build_metadata(os.getcwd(), 'm5stack-coreS3', cache=True)["targets"]))
                     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.....
  File "/Users/mah/.platformio/penv/lib/python3.13/site-packages/platformio/platform/_run.py", line 73, in _run_scons
    os.path.join(scons_dir, "scons.py"),
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen posixpath>", line 77, in join
TypeError: expected str, bytes or os.PathLike object, not NoneType
`````
- close project, restart VSCode - same thing: it would stall on trying to run `scons.py`
- as for the first error, googling for the `ssl.SSLCertVerificationError` led my to [stackoverflow](https://stackoverflow.com/questions/44649449/brew-installation-of-python-3-6-1-ssl-certificate-verify-failed-certificate/44649450#44649450) which mentions that Brew Python for some reason does not run the `Install\ Certificates.command` 
- this smelled, and despite this was not Brew Python, I did run ` /Applications/Python\ 3.13/Install\ Certificates.command`, and retried
- while the `ssl.SSLCertVerificationError` splat was gone, the `scons.py` problem persisted
- It turned out the directoy `.platformio/packages/tools-scons`  exist BUT was empty - I suspice the SSL error happened during installation of this package and ended half-done; thereafter pioarduino would see the tool-scons directory, assume its installed, called scons.py therein and peng
- at this point I exited VSCode, uninstalled pioarduino and wiped `.platformio` again, and low and behold, the `tool-scons`  directory was eventually populated

From thereon it was smooth sailing - project configured and built fine.

I also noticed that "uninstalling" an extension from VSCode does NOT remove the corresponding `.vscode/exentsions/<extension-subdir>`
Despite uninstalling Platformio and Pioarduino extensions I found there:
`````
pioarduino.pioarduino-ide-1.0.6
platformio.platformio-ide-3.3.3-darwin-arm64
platformio.platformio-ide-3.3.4-darwin-arm64
`````
I wiped all of those for good measure as I noticed what @tschissler observed above - PlatformIO reappearing out of the blue.

TLDR; 
- make sure to run the `Install Certificates` command for your Python version
- dont just uninstall the extensions, wipe the corresponding dirs under `.vscode/exentsions/`
- watch for empty dirs under `.platformio` as these could be leftovers from an aborted install attempt
